docs/developer-feature-tutorial.md

```diff
@@ -1,59 +1,92 @@
-## Tutorial - Adding typing indicator
+# Developer Feature Tutorial
+This tutorial goes through the process of writing a new application feature to Zulip Terminal for first time contributers. 
 
-This tutorial shows how typing indicator was implemented in the client. The process for adding a new feature to zulip terminal varies greatly depending on the feature. This tutorial is intended to make you familiar with the general process.
+The process for adding a new feature to zulip terminal varies greatly depending on the feature. Hence, this tutorial is intended to make you familiar with the general process, using an example.
 
-Since the typing indicator data for the other user in pm cannot be generated locally, it should be received from the client.
+Before diving into this, we recommend you go through the Zulip's brilliant [Git Guide](https://zulip.readthedocs.io/en/latest/git/index.html), as we follow a **rebase-oriented workflow**, and our [commit message giudelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html).
```

Comment:
The second link in this line speaks about commit discipline as a whole, so it would be more appropriate if you can use a better alias than "commit message guidelines" (typo - "guidelines" as well). There's also no need to use "the" before "Zulip's".